### DISCUSSION TITLE
How to voice input on android[Question]

### DISCUSSION THREAD

**Discussion by jkloip** [flet-dev/flet#2221] [Q&A]
### Question

On Windows, I use pyaudio and wave to develop voice input and recognition functions, but these two packages cannot be used on Android. How do I do this on Android? I am currently developing a two-way translation software that allows both users to input and recognize text through voice. During this time, I used Flet to develop the Windows version. I like using Flet very much, but I cannot use the microphone on Android. It's bothering me and I don't know how to solve it? Can I just switch to Flutter? Please give me any help or tips, I would be grateful.

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by burhansvural** [comment]
> ### Question
> On Windows, I use pyaudio and wave to develop voice input and recognition functions, but these two packages cannot be used on Android. How do I do this on Android? I am currently developing a two-way translation software that allows both users to input and recognize text through voice. During this time, I used Flet to develop the Windows version. I like using Flet very much, but I cannot use the microphone on Android. It's bothering me and I don't know how to solve it? Can I just switch to Flutter? Please give me any help or tips, I would be grateful.
> 
> ### Code sample
> _No response_
> 
> ### Error message
> _No response_
> 
> ### ------------------------------------------------------
> * [x]  I have searched for answers to my question both in the issues and in previous discussions.

https://stackoverflow.com/questions/72659181/is-there-any-way-to-get-microphone-input-on-android-using-python

**Reply by 09u2h4n** [reply]
Flet 0.20.0 came.

**Reply by burhansvural** [reply]
@09u2h4n Oğuzhan kardeş, if you want to see more changes, you can install the -pre version.

**Reply by 09u2h4n** [reply]
I replied to the wrong comment 😅

**Comment by ndonkoHenri** [comment]
The upcoming Flet version 0.20.0 contains a new control that will satisfy your needs : `AudioRecorder`.

**Comment by 09u2h4n** [comment]
Flet 0.20.0 came
